AdminServer/appscale/admin/instance_manager/instance_manager.py

```diff
@@ -0,0 +1,660 @@
+""" Fulfills AppServer instance assignments from the scheduler. """
+import logging
+import math
+import json
+import os
+import psutil
+import signal
+import urllib2
+
+from tornado import gen
+from tornado.ioloop import IOLoop, PeriodicCallback
+from tornado.locks import Lock as AsyncLock
+
+from appscale.admin.constants import UNPACK_ROOT
+from appscale.admin.instance_manager.constants import (
+  API_SERVER_LOCATION, API_SERVER_PREFIX, APP_LOG_SIZE, BACKOFF_TIME,
+  BadConfigurationException, DASHBOARD_LOG_SIZE, DASHBOARD_PROJECT_ID,
+  DEFAULT_MAX_APPSERVER_MEMORY, FETCH_PATH, GO_SDK, INSTANCE_CLASSES,
+  JAVA_APPSERVER_CLASS, MAX_API_SERVER_PORT, MAX_INSTANCE_RESPONSE_TIME,
+  MONIT_INSTANCE_PREFIX, NoRedirection, PIDFILE_TEMPLATE, PYTHON_APPSERVER,
+  START_APP_TIMEOUT, STARTING_INSTANCE_PORT, VERSION_REGISTRATION_NODE)
+from appscale.admin.instance_manager.instance import (
+  create_java_app_env, create_java_start_cmd, create_python_app_env,
+  create_python27_start_cmd, get_login_server, Instance)
+from appscale.admin.instance_manager.stop_instance import stop_instance
+from appscale.admin.instance_manager.utils import setup_logrotate
+from appscale.common import appscale_info, monit_app_configuration
+from appscale.common.async_retrying import retry_data_watch_coroutine
+from appscale.common.constants import (
+  APPS_PATH, GO, HTTPCodes, JAVA, MonitStates, PHP, PYTHON27, VAR_DIR,
+  VERSION_PATH_SEPARATOR)
+from appscale.common.monit_interface import DEFAULT_RETRIES, ProcessNotFound
+from appscale.common.retrying import retry
+
+logger = logging.getLogger('appscale-instance-manager')
```

Comment:
I'm not clear on the use of logging vs. logger, may be worth a check to ensure it is as intended.

Reply:
`logging` is a shortcut to the root logger. `logger` here is a custom logger that can have a different format and level than the root one.

Some libraries use the root logger to log messages, and we were recently using `logging` everywhere. In one case, this meant that when we enabled debug-level logging, we got super verbose debug logs from the kazoo library.

Recently, we've been using a custom logger to avoid this problem. The references to `logging` in this module were left over from when I moved the code from the app_manager_server script. In this case, it won't make a difference since they both happen to have the same level and format, and the user doesn't yet have a way to change it.

However, it's helpful that you pointed out those cases since we should be more consistent here, and we will likely make this more configurable. I've created a new issue for it (https://github.com/AppScale/appscale/issues/2893) that I will address for the whole appscale-admin package after this PR goes in.